processing/src/main/java/org/apache/druid/query/timeboundary/TimeBoundaryQueryQueryToolChest.java

```diff
@@ -224,4 +226,40 @@ public Result<TimeBoundaryResultValue> apply(Object input)
       }
     };
   }
+
+  @Override
+  public RowSignature resultArraySignature(TimeBoundaryQuery query)
+  {
+    if (query.isMinTime() || query.isMaxTime()) {
+      RowSignature.Builder builder = RowSignature.builder();
+      String outputName = query.isMinTime() ?
+                          query.getContextValue(TimeBoundaryQuery.MIN_TIME_ARRAY_OUTPUT_NAME, TimeBoundaryQuery.MIN_TIME) :
+                          query.getContextValue(TimeBoundaryQuery.MAX_TIME_ARRAY_OUTPUT_NAME, TimeBoundaryQuery.MAX_TIME);
+      return builder.add(outputName, ColumnType.LONG).build();
+    }
+    return super.resultArraySignature(query);
```

Comment:
Should we throw a more meaningful UOE here instead of relying on the method in the super class?
```suggestion
    throw new UOE("TimeBoundaryQuery with unsupported bounds [%s]", query.getBound());
```

similar comment for the method below

Reply:
Actually, reading the implementation of `TimeBoundaryQuery#mergeResults`, it looks like `bound` being set to something other than min or max is supported and it is handled by returning both min and max for the result.

Reply:
Yes, I'll update the exception to something more meaningful.

Regarding, the handling of min-max part. Yes, the native query returns both min-max if no bounds given. But in the first implementation, it was becoming a bit dirty to map the min-max results of a SQL query so decided to throw it for a start. Will take another stab to see if we can accommodate `SELECT min(__time), max(__time) from ds` and `SELECT max(__time), min(__time) from ds`

Reply:
updated the PR desc with the min-max query limitation